### PR TITLE
Using PredicateBuilder class for querying Users and Groups

### PR DESCRIPTION
**Pull request by CecilyK**

### CODE REVIEW

File: Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs [hunk 1]
```diff
@@ -186,32 +165,34 @@ public override Task<Resource[]> QueryAsync(IQueryParameters parameters, string
...
-                                users.Where(
-                                    item =>
-                                       item.Active == bool.Parse(andFilter.ComparisonValue)).ToList();
+                            bool active = bool.Parse(andFilter.ComparisonValue);
+                            predicateAnd = predicateAnd.And(p => p.Active == active);
+
                         }
 
-                        // LastModified filter
+                        //
```
Comment (marcusca10) [inline Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs:173]: @CecilyK could you update this comment, the value should be: //LastModified filter

File: Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs [hunk 2]
```diff
@@ -222,17 +203,20 @@ public override Task<Resource[]> QueryAsync(IQueryParameters parameters, string
...
 
             if (parameters.PaginationParameters != null)
             {
                 int count = parameters.PaginationParameters.Count.HasValue ? parameters.PaginationParameters.Count.Value : 0;
-                return Task.FromResult(results.Take(count).ToArray());
+                return Task.FromResult((results.Take(count).ToArray()));
             }
             else
-                return Task.FromResult(results.ToArray());
+                return Task.FromResult((results.ToArray()));
```
Comment (marcusca10) [inline Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs:219]: @CecilyK clean up the additional parentheses ;)

File: Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs [hunk 3]
```diff
@@ -222,17 +203,20 @@ public override Task<Resource[]> QueryAsync(IQueryParameters parameters, string
...
                 }
+
+                results = this.storage.Users.Values.Where(predicate.Compile());
             }
 
             if (parameters.PaginationParameters != null)
             {
                 int count = parameters.PaginationParameters.Count.HasValue ? parameters.PaginationParameters.Count.Value : 0;
-                return Task.FromResult(results.Take(count).ToArray());
+                return Task.FromResult((results.Take(count).ToArray()));
```
Comment (marcusca10) [inline Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs:216]: @CecilyK clean up the additional parentheses ;)

**Review by marcusca10 — CHANGES_REQUESTED** [review]
@CecilyK just some code cleanup and we're good

File: Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs (hunk 1 above)
Comment (CecilyK) [reply Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs]: Done :)

File: Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs (hunk 2 above)
Comment (CecilyK) [reply Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs]: Done :)

File: Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs (hunk 3 above)
Comment (CecilyK) [reply Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs]: Done :)

**Review by marcusca10 — APPROVED** [review]
Thanks

**Review by boscoMW — APPROVED** [review]
LGTM

**Pull request merged**